### PR TITLE
Working branch

### PR DESCRIPTION
**Pull request by john-s-m**
Serialization code is complete...   Need more tests though

### CODE REVIEW

File: cacheMgr/cacheMgr.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+type SafeMapAccess interface {
+	Reader( key int ) interface{}
+	Writer( key int, value interface{} )
+	Inserter( key int, ci *CacheItem )
+	serializer()
+}
+
+func NewCacheItem( val interface{}, goRtn bool ) *CacheItem {
```
Comment (hms) [inline cacheMgr/cacheMgr.go:35]: goRtn parameter name isn't descriptive.  Isn't it really 'needsSerialization'

File: cacheMgr/cacheMgr.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	SharedMap map [int]CacheItem
+	readCh chan interface{}
+	readKeyCh chan int
+	writeCh chan CacheKeyValue
+	insertCh chan CacheKeyValue
+}
+
+type SafeItemAccess interface {
+	Reader() interface{}
+	Writer( interface{} )
```
Comment (hms) [inline cacheMgr/cacheMgr.go:24]: Should return an error status

File: cacheMgr/cacheMgr.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+	keyVal := CacheKeyValue{ key, val }
+	cm.insertCh <- keyVal
+}
+
+func (cm CacheMap) serializer () {
+	var keyVal CacheKeyValue
+	var key int
+	for {
+		select {
+		case keyVal = <-cm.writeCh:
```
Comment (hms) [inline cacheMgr/cacheMgr.go:96]: Isn't this really an update?

**Review by hms — CHANGES_REQUESTED** [review]
Good start

File: cacheMgr/cacheMgr.go (hunk 1 above)
Comment (john-s-m) [reply cacheMgr/cacheMgr.go:35]: done

File: cacheMgr/cacheMgr.go (hunk 2 above)
Comment (john-s-m) [reply cacheMgr/cacheMgr.go:24]: done

File: cacheMgr/cacheMgr.go (hunk 3 above)
Comment (john-s-m) [reply cacheMgr/cacheMgr.go:96]: done

**Comment by john-s-m** [comment]
changes made...  By the way I do need readChKey...

**Pull request merged**